mpa/modules/hooks/auxiliary_hooks.py

```diff
@@ -13,23 +13,48 @@
 # and limitations under the License.
 
 from __future__ import annotations
+from abc import ABC, abstractmethod
 from typing import Union
 
 import torch
 
 
-class EigenCamHook:
-    def __init__(self, module: torch.nn.Module) -> None:
+class BaseAuxiliaryHook(ABC):
+    def __init__(self, module: torch.nn.Module, _fpn_idx: int = 0) -> None:
```

Comment:
The functionality of this class is not Auxiliary anymore.
How about "RecordingForwardHook", "ModuleRecordingHook", etc?
BTW, don't use _xxxx name for public parameters. "_fpn_idx"???

Reply:
Yes, as you said recording hook is more appropriate. The hook name changed and `_` removed from the parameter name.